src/lib/Types/Imp.hs

```diff
@@ -33,16 +33,15 @@ import Util (IsBool (..))
 
 import Types.Primitives
 
--- We use AtomName because we convert between atoms and imp
--- expressions without chaning names. Maybe we shouldn't do that.
-type ImpName = Name AtomNameC
+type ImpName = Name ImpNameC
 
 type ImpFunName = Name ImpFunNameC
 data IExpr n = ILit LitVal
              | IVar (ImpName n) BaseType
+             | IPtrLit (Name PtrNameC n) PtrType
```

Comment:
Well it's a pointer var, not a pointer literal, right?

Reply:
Good point(er). I'll rename it.